### ISSUE TITLE
DB Error - duplicate entry for 'url' in 'fcontact'

### ISSUE THREAD

**Issue by rabuzarus** [friendica/friendica#4997]
From my logfile.out

`dba.php:474:p  DB Error 1062: Duplicate entry 'https://libranet.de/profile/halle_saale' for key 'url'
q, Diaspora::addFContact, Diaspora::personByHandle, Diaspora::key, Diaspora::decodeRaw, dfrn_dispatch_private, dfrn_notify_post
INSERT INTO `fcontact` (`url`,`name`,`photo`,`request`,`nick`,`addr`, `guid`, `batch`, `notify`,`poll`,`confirm`,`network`,`alias`,`pubkey`,`updated`) VALUES ('https://libranet.de/profile/halle_saale','HalleSpektrum (inoffiziell)','https://libranet.de/photo/98931883545aed89003c244336103881-4.png?ts=1525509397','','halle_saale','halle_saale@libranet.de','0b6b25a8965aed8353609ac397183001','https://libranet.de/receive/public','https://libranet.de/receive/users/0b6b25a8965aed8353609ac397183001','https://libranet.de/dfrn_poll/halle_saale','','dspr','https://libranet.de/~halle_saale','-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEA222OP1Ss+a0xt1QENCSl\ng4V894VZq+iJri8xHTdh1Piufp0EqOrJokixdAfbSuAlK4rChZigITJCntSp5epY\n1RdWzoZ/CBmDU7o2zCWFqZPQQGL0e4q5gvgT1Wx+HC3PqBveYkqxbPKpQI1Cp7Ut\nfWJ9nYGIo7+cWKWRlDgkawrt2adyDkDSY6DpD3uLlaYujESfIegHpitNYKoTYXPa\nVgkjLLc62v7XmHn59lEkPvF0UWiGaXj2kQFjHaXm9nLnf3v03qDHKypcqB+Uazv8\nsRLigj/VTyDpYjKIoEotWB4KGI5sr+uU1d/PcMYpKJ+BzYdH6i6mi3QxzAD2DF99\nJfFXNpOlsDViJbD83ubDibFd5hBE5iTqhp7IcN6FRGJVGumIrkK0I28oWWL3ox7c\ni4xg12uKGZFBbFWEcTYcI41kEQfmiuEque6+ZBKWPPKDwmZ3eziElt9Z3axezBJc\nkURNYh0jil2KxsNgZ9EvbRA5IO4VS/WFIBy8aCL69EJ0UOw44pelVmm/OGLhmtv9\n0VNy+mIVXz8hI9fTSYyhP4dkBmU0yPOw7dXFea5Y064dQAEeSQQhTni9nEkS46c7\nm/rb7RUV/WMsKV6vj+H6Q1hXD0649IAsjchTw8BCwbUuciDb58pIMtRw/oPh/CRj\nBdLh+n9T88z6zEP04lbXdk0CAwEAAQ==\n-----END PUBLIC KEY-----\n','2018-05-08 09:35:04')`

### Friendica version you encountered the problem
2018.05-rc

### PHP version
7.0
### SQL version
mysql  Ver 15.1 Distrib 10.0.32-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

**Comment by annando** [comment]
How often does this occur? I would like to close it when it only happens a few times a day.

**Comment by annando** [comment]
With the same? This is weird.

**Comment by AlfredSK** [comment]
From my logfile from yesterday:
````
grep -Ei 'db error 1062' logs/friend.log.1 | wc -l
536
````
Example:
````
2018-05-10T04:14:33Z@wrk5af3c723d9fb83.67182935 [NORMAL]:dba.php:539:e  DB Error 1062: Duplicate entry '6649-foss' for key 'PRIMARY'
````
I guess these duplicates are about tags not URLs. So, maybe that's another issue?

**Comment by annando** [comment]
This is some other problem that we cannot solve easily. Depending on the database collation an "u" could be the same like "ü" or "ü" could be the same like "ue" for example. So if the system wants to store the hash tag "müllabfuhr" it could have a problem if there had been a tag "muellabfuhr".

**Comment by AlfredSK** [comment]
I refined the regex for grep:
````
grep -Ei 'db error 1062.*url' logs/friend.log.1 | wc -l
6
````
... only 6 entries (for two different URLs).
````
2018-05-09T15:22:12Z@wrk5af31224b39508.43498706 [NORMAL]:dba.php:474:p  DB Error 1062: Duplicate entry 'https://gourmetkater.com/~gourmetkater/profile/gourmetkater' for key 'url'
2018-05-10T02:49:30Z@log5af3b3370e4456.30698980 [NORMAL]:dba.php:474:p  DB Error 1062: Duplicate entry 'https://pod.aevl.us/u/lele0x' for key 'url'
````

**Comment by MrPetovan** [comment]
Closed via #5018
